python/tchannel/theader.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+
+class RetryType(object):
+    NEVER = 'n'
+    CONNECTION_ERROR = 'c'
+    TIMEOUT = 't'
+    CONNECTION_ERROR_AND_TIMEOUT = 'ct'
+    TIMEOUT_AND_CONNECTION_ERROR = 'tc'
+    DEFAULT = 'c'
+
+
+class ArgSchemeType(object):
```

Comment:
Wouldn't be the worst idea to use `enum.Enum` here and in `RetryType`. You'll be able to use the string to automatically cast it into the ArgSchemeType object using `ArgSchemeType('http')` by checking `isinstance(s, basestring)` first.

Also, this seems completely disconnected from the other `ArgScheme` stuff.